src/Tizen.Network.Nsd/Tizen.Network.Nsd/DnssdService.cs

```diff
@@ -217,15 +219,68 @@ public IPAddressInformation IP
                 return IPAddressInstance;
             }
         }
-
-        private void GetTxtRecord(out ushort length, out byte[] value)
+        
+        /// <summary>
+        /// Returns TXT records in the form of key and value.
+        /// </summary>
+        /// <remarks>
+        /// After adding the TXT record call for get TXT record.
+        /// returns empty dictionary in case of empty txt record
+        /// </remarks>
+        /// <since_tizen> 9 </since_tizen>
+        /// <feature>http://tizen.org/feature/network.service_discovery.dnssd</feature>
+        /// <exception cref="NotSupportedException">Thrown when DNS-SD is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when any other error occurred.</exception>
+        public IDictionary<string, string> GetTxtRecord()
+        {
+            var map = new Dictionary <string, string>();
+            
+            byte[] data = GetRawTxtRecord();
+            if (data.Length > 0)
+            {
+                var index = 0;
+                while (index < data.Length)
+                {
+                    var txtLen = data[index++];
+                    if (txtLen > data.Length)
+                        Log.Debug(Globals.LogTag, "Invalid data");
+
+                    string record = Encoding.UTF8.GetString(data, index, txtLen);
+                    string[] list = record.Split('=');
+                    map.Add(list[0], list[1]);
+                    index += txtLen;
+                }
+            }
+            return map;
+        }
+        
+        /// <summary>
+        /// Returns all raw TXT records.
+        /// </summary>
+        /// <remarks>
+        /// After adding the TXT record call for get TXT record.
+        /// </remarks>
+        /// <since_tizen> 9 </since_tizen>
+        /// <feature>http://tizen.org/feature/network.service_discovery.dnssd</feature>
+        /// <exception cref="NotSupportedException">Thrown when DNS-SD is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when any other error occurred.</exception>
+        public byte[] GetRawTxtRecord()
         {
-            int ret = Interop.Nsd.Dnssd.GetAllTxtRecord(_serviceHandle, out length, out value);
+            byte[] value = Array.Empty<byte>(); 
+            int ret = Interop.Nsd.Dnssd.GetAllTxtRecord(_serviceHandle, out ushort length, out IntPtr data);
             if (ret != (int)DnssdError.None)
             {
                 Log.Error(Globals.LogTag, "Failed to get the TXT record, Error: " + (DnssdError)ret);
                 NsdErrorFactory.ThrowDnssdException(ret);
             }
+            if (length > 0)
+            {
+                byte[] managedArray = new byte[length];
+                Marshal.Copy(data, managedArray, 0, length);
+                value = new byte[length];
+                managedArray.CopyTo(value, 0);
```

Comment:
Why not directly return managedArray?
```suggestion
                return managedArray;
```

Reply:
review comments applied